components/contactUs.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useState, useRef, useEffect } from "react";
+import { RenderMsg } from "./renderMsg";
+
+type FormState = {
+  name: string;
+  email: string;
+  message: string;
+};
+
+export const ContactFrom = () => {
+  const [nameErr, setNameErr] = useState("");
+  const [emailErr, setEmailErr] = useState("");
+  const [messageErr, setMessageErr] = useState("");
+  const [state, setstate] = useState<FormState>({
+    name: "",
+    email: "",
+    message: "",
+  });
+
+  const nameInput = useRef<HTMLInputElement>(null);
+  const emailInput = useRef<HTMLInputElement>(null);
+  const messageInput = useRef<HTMLTextAreaElement>(null);
+
+  useEffect(() => {
+    if (nameInput.current != null) {
+      nameInput.current.focus();
+    }
+  }, []);
+
+  const setFocus = (
+    ref:
+      | React.RefObject<HTMLInputElement>
+      | React.RefObject<HTMLTextAreaElement>
+  ) => {
+    if (ref.current != null) {
+      ref.current.focus();
+    }
+  };
+
+  const handleChange = (
+    event:
+      | React.ChangeEvent<HTMLInputElement>
+      | React.ChangeEvent<HTMLTextAreaElement>
+  ) => {
+    setstate({ ...state, [event.target.name]: event.target.value });
+  };
+
+  const formIsValid = () => {
+    let validRegex =
+      /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/;
+
+    if (state.name.length === 0) {
+      setNameErr("This field is required");
+      setFocus(nameInput);
+      return 0;
```

Comment:
Why return `0` instead of `false`? If you have a function that returns a "yes or no" answer - use a `boolean`. You will use the result of this function as a condition of an if statement, so its type will get coerced anyway, but this is just adding an unnecessary layer of abstraction.